content/1_docs/2_cookbook/0_forms/0_basic-contact-form/recipe.txt

```diff
@@ -0,0 +1,202 @@
+Title: A basic email contact form
+
+----
+Description: Learn how create a basic contact form that sends an HTML or plain text email
+
+----
+Text:
+
+Form handling is always a bit of a pain. In this recipe we'll therefore go through a basic example that you can then extend according to your needs.
+
+This example consists of several files
+
+- a contact page
+- a (glossary: template text: template) with the form
+- a (glossary: controller text: controller) with the form logic that handles all the "heavy lifting"
+- a plain text and an HTML email (glossary: snippet text: snippet)
+
+## The contact page
+
+(link: docs/guide/content/pages text: Create a contact page) with a `contact.txt` file in it. The minimum we want in the content file is a title, the rest is up to you. You could, for example, store an introductory text or some variables like the email address to send the email to or whatever you think wise.
+
+For use in the Panel, you can also create a (glossary: blueprint text: blueprint) for the page. We skip this step here.
+
+## The contact template
+
+Our `contact.php` template contains the form and will display error messages if something goes wrong or a success message if the form was successfully sent.
+
+```php "/site/templates/contact.php"
+
+<?php snippet('header') ?>
+    <main class="main">
+        <h1><?= $page->title()->html() ?></h1>
+
+        <?php if($success): ?>
+        <div class="alert success">
+            <p><?= $success ?></p>
+        </div>
+        <?php else: ?>
+            <?php if (isset($alert['error'])): ?>
+                <div><?= $alert['error'] ?></div>
+            <?php endif ?>
+        <form method="post" action="<?= $page->url() ?>">
+            <div class="honeypot" style="display:none">
```

Comment:
My 2c, I would rather position the honeypot off-screen rather than with a `display: none` that some robots are likely to detect.

Reply:
Good point.